db/migrate/20220826034057_create_katalyst_navigation_menus.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class CreateKatalystNavigationMenus < ActiveRecord::Migration[7.0]
+  def change
+    create_table :katalyst_navigation_menus do |t|
+      t.string :title
+      t.string :slug, index: true
+
+      t.references :current_version
+      t.references :latest_version
+
+      t.timestamps
+    end
+
+    create_table :katalyst_navigation_menu_versions do |t|
+      t.belongs_to :parent, foreign_key: { to_table: :katalyst_navigation_menus }, null: false
+      t.json :nodes
```

Comment:
`t.jsonb` instead? allows native querying on the keys.

Pros/cons: https://railsexamples.com/postgresql-json/

Reply:
It's not compatible with sqlite, which we're using for tests. I don't really want to limit us to Postgres as this is in theory a gem we might reuse in lots of different places. The data being stored is pretty small, only 2 key/values per item in the tree.